storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -266,12 +276,53 @@ public void storeFinalizedBlocks(final Collection<SignedBeaconBlock> blocks) {
       expectedRoot = block.getParentRoot();
     }
 
+    // Check block signatures are valid for blocks except the genesis block
+    Collection<SignedBeaconBlock> nonGenesisBlocks =
+        blocks.stream()
+            .filter(block -> !block.getSlot().equals(UInt64.ZERO))
+            .collect(Collectors.toList());
+    checkArgument(
+        batchVerifyHistoricalBlockSignatures(nonGenesisBlocks), "Block signatures are invalid");
```

Comment:
We could just check if the last block in `sorted` is the genesis block and if it isn't use the original list, if it is just use `sorted.subList` to exclude it without having to copy all the items.

Reply:
Ah forgot that method existed. Will update.

Reply:
Done.